src/App.js

```diff
@@ -1,106 +1,134 @@
+        <SignOut />
       </header>
+
+      <section>
+        {user ? <ChatRoom /> : <SignIn />}
+      </section>
+
     </div>
   );
 }
 
+function SignIn() {
 
-function signIn() {
   const signInWithGoogle = () => {
     const provider = new firebase.auth.GoogleAuthProvider();
     auth.signInWithPopup(provider);
-  };
+  }
 
-  return(
-    <button onClick = {signInWithGoogle}> Sign in with Google </button>
+  return (
+    <>
+      <button className="sign-in" onClick={signInWithGoogle}>Sign in with Google</button>
+      <p>Do not violate the community guidelines or you will be banned for life!</p>
+    </>
   )
 
-};
+}
 
-function signOut() {
+function SignOut() {
   return auth.currentUser && (
-    <button onClick = {() => auth.signOut()}> Sign Out</button>
+    <button className="sign-out" onClick={() => auth.signOut()}>Sign Out</button>
   )
+}
 
-};
 
+function ChatRoom() {
+  const dummy = useRef();
+  const messagesRef = firestore.collection('messages');
+  const query = messagesRef.orderBy('createdAt').limit(25);
 
+  const [messages] = useCollectionData(query, { idField: 'id' });
 
+  const [formValue, setFormValue] = useState('');
 
-function chatRoom() {
-  const messagesRef = firestore.collection('messages');
-  const query = messagesRef.orderBy('createdAt').limit(25);
 
-  const [messages] = useCollectionData(query, {idField: 'id'});
+  const sendMessage = async (e) => {
+    e.preventDefault();
 
-  return(
-    <>
-    <div>
-      {messages && messages.map(msg => <ChatMassage key={msg.id} massage={msg} />)}
-    </div>
-    
-    <form>
-      <input />
-      <button type = 'submit'> Submit </button>
+    const { uid, photoURL } = auth.currentUser;
 
-    </form>
-    </>
-  )
+    await messagesRef.add({
+      text: formValue,
+      createdAt: firebase.firestore.FieldValue.serverTimestamp(),
+      uid,
+      photoURL
+    })
 
-};
+    setFormValue('');
+    dummy.current.scrollIntoView({ behavior: 'smooth' });
+  }
 
+  return (<>
+    <main>
 
-function ChatMassage(props) {
-  const {text, uid, photoURL} = props.message;
-  const messageClass = uid === auth.currentUser.uid ? 'sent' : 'received' ;
+      {messages && messages.map(msg => <ChatMessage key={msg.id} message={msg} />)}
 
+      <span ref={dummy}></span>
 
-  return (
-    <div className = {`message ${messageClass}`}>
-      <img src = {photoURL} />
-      <p> {text} </p>
-      
+    </main>
+
+    <form onSubmit={sendMessage}>
+
+      <input value={formValue} onChange={(e) => setFormValue(e.target.value)} placeholder="say something nice" />
+
+      <button type="submit" disabled={!formValue}>🕊️</button>
 
+    </form>
+  </>)
+}
+
+
+function ChatMessage(props) {
+  const { text, uid, photoURL } = props.message;
+
+  const messageClass = uid === auth.currentUser.uid ? 'sent' : 'received';
+
+  return (<>
+    <div className={`message ${messageClass}`}>
+      <img src={photoURL || 'https://api.adorable.io/avatars/23/abott@adorable.png'} />
```

Comment:
My linter keeps asking me to add an alt into the img tag, could you do that?
`alt="profile pic`